### PR TITLE
initial docker files

### PR DESCRIPTION
**Pull request by xeroc**
These docker files build and start the bitshares node according to the default config file in `docker/` folder.

I would like to merge this with `master` and `develop` so that I can integrate continuous integration with docker hub.

The only thing **open to discussion** is if the `bitshares-core` container should actually **run** the witness node already, or if we should use `bitshares-core` as a dummy container that only builds the repo for other containers to use and run witness_node, or cli_wallet or delayed_node. Thoughts?

### CODE REVIEW

**Comment by xeroc** [comment]
After some discussion with @billbutler, it turns out that the `CMD` provided in the DockerFile is just a default and can be easily overwriten using docker-compose. That means, I would propose to keep the docker container to run the default witness node by default .. other use cases (like witnessing etc) can be done separately.

**Comment by oxarbitrage** [comment]
https://github.com/bitshares/bitshares-core/pull/376

**Comment by oxarbitrage** [comment]
thank you for working on this, let me ask you because i don't know much about docker. does your pull replaces https://github.com/bitshares/bitshares-core/pull/376 ?

**Comment by dotneet** [comment]
@oxarbitrage 
This PR and #376 have different purpose fundamentally.
#367 provide an ability to build "bitshares-core" easily.
This PR provide an ability to run "bitshares-core" in docker container easily.
Ideally both should be supported i think.

**Comment by xeroc** [comment]
This PR adds a basic dockerfile that builds bitshares in a container and also also users to deploy a 'default node'.

The whole idea is to add build automation to the whole repo (including all branches) through the docker cloud ..

Once set up, we can 
* easily compile+test ever branch and pull request
* easily (trivially) deploy new nodes for bitshares

Later this week, i want to add some kore configuration options the the launch script so that

* i can configute 'plugins'
* use custom config files (e.g. for witnessing)
* add docker compose to be able to easily run the cli-wallet and trusted node

Basically, i want to have a setup where exchanges only download a docker-compose YAML file and run `docker-compose up` to have all they need.

Backend updates would be trivial than too.


The reason for docker cloud automated building simply is transparency .. We dont push containers but they are built by docker cloud for everyone to investigate on

**Comment by oxarbitrage** [comment]
@xeroc i think it will be better if we can send pulls for this to `develop` branch so they will be merged to `master` in the next release.

**Comment by xeroc** [comment]
Ideally, I'd like to have a docker container for the current release already. However, I cannot merge this into an existing tag. But we could merge this with master which represents the latest tag and later just relabel the contain with the release tag number ...

I would really prefer this to be merged into `master` (and `develop`).

**Comment by abitmore** [comment]
@xeroc we can always merge it to master later, although maybe manually. I think it's good to have it in develop branch as well, so merging to develop first is a good choice.

**Comment by xeroc** [comment]
ok .. closing this PR to open up a new one

**Comment by abitmore** [comment]
I'm not familiar with Docker, but I think the original file should be kept in some form if it's useful. Ideally, we should have multiple Docker configurations for different purposes simultaneously.

But the way, no need to close a PR if you only need to change the base branch, as that can be done by selecting another base branch when editing the subject.

**Comment by xeroc** [comment]
> Ideally, we should have multiple Docker configurations for different purposes simultaneously.

This can be achieved with docker-compose easily:

```
 trusted:
  image: .......
  environment:
   BITSHARESD_ARGS: --help
  volumes:
   - ./docker/conf/:/etc/bitshares/
  command: "cat /etc/bitshares/config.ini"
```

> But the way, no need to close a PR if you only need to change the base branch, as that can be done by selecting another base branch when editing the subject.

Ha .. learn something new every day :D
